test/rekt/features/broker_deleted_recreated.go

```diff
@@ -131,6 +131,63 @@ func BrokerAuthSecretDoesNotExist() *feature.Feature {
 	return f
 }
 
+// BrokerExternalTopicDoesNotExist tests that a broker can be deleted without the Topic existing.
+func BrokerExternalTopicDoesNotExist() *feature.Feature {
+	f := feature.NewFeatureNamed("delete broker with non existing Topic")
+
+	brokerName := feature.MakeRandomK8sName("broker")
+	configName := feature.MakeRandomK8sName("config")
+	topicName := feature.MakeRandomK8sName("topic") // A k8s name is also a valid topic name.
+
+	f.Setup("create broker config", brokerconfigmap.Install(
+		configName,
+		brokerconfigmap.WithBootstrapServer(testpkg.BootstrapServersPlaintext),
+	))
+
+	f.Setup("install broker", broker.Install(
+		brokerName,
+		append(
+			broker.WithEnvConfig(),
+			broker.WithConfig(configName),
+			broker.WithAnnotations(
+				map[string]interface{}{
+					kafkabroker.ExternalTopicAnnotation: topicName,
+				}))...))
+
+	f.Assert("delete broker", featuressteps.DeleteBroker(brokerName))
+
+	return f
+}
+
+// BrokerExternalTopicAuthSecretDoesNotExist tests that a broker can be deleted without the Secret or Topic existing.
+func BrokerExternalTopicAuthSecretDoesNotExist() *feature.Feature {
+	f := feature.NewFeatureNamed("delete broker with non existing Secret or Topic")
+
+	brokerName := feature.MakeRandomK8sName("broker")
+	configName := feature.MakeRandomK8sName("config")
+	topicName := feature.MakeRandomK8sName("topic") // A k8s name is also a valid topic name.
+
+	f.Setup("create broker config", brokerconfigmap.Install(
+		configName,
+		brokerconfigmap.WithBootstrapServer(testpkg.BootstrapServersPlaintext),
+		brokerconfigmap.WithAuthSecret("does-not-exist"),
+	))
+
```

Comment:
Wouldn't it make sense to create the topic here, just to be sure, that that's not the reason the broker got deleted again?

Reply:
```
// BrokerExternalTopicAuthSecretDoesNotExist tests that a broker can be deleted without the Secret or Topic existing.
```